### PR TITLE
feat: Brazilian standard require idNumber, back document and email fo…

### PR DESCRIPTION
**Pull request by ailsoncgt**
kyc validation

### CODE REVIEW

File: fiatconnect-api.md [hunk 1]
```diff
@@ -2226,12 +2226,15 @@ A KYC schema containing personal data about a user, as well as documents such as
 		postalCode?: `string`
 	},
 	phoneNumber: `string`,
+	idNumber?: `string`,
+	email?: `string`,
 	selfieDocument: `string`,
-	identificationDocument: `string`
+	identificationDocument: `string`,
+	identificationDocumentBack?: `string`,
```
Comment (cajubelt) [inline fiatconnect-api.md:2233]: if you require `idNumber`, `email`, and `identificationDocumentBack`, then they shouldn't be optional fields. The question mark indicates that the field is optional.

Comment (cajubelt) [reply fiatconnect-api.md:2233]: Also, I recommend that you create a new KYC schema for these fields, rather than tacking them onto an existing one. Adding a new KYC schema is a non-breaking change, since clients can incrementally add support for new schemas without breaking their existing integrations. However, changing an existing KYC schema is generally a breaking change, and something to avoid without some compelling reason for it

File: fiatconnect-api.md [hunk 2]
```diff
@@ -2233,6 +2234,37 @@ A KYC schema containing personal data about a user, as well as documents such as
...
+	phoneNumber: `string`,
+	idNumber: `string`,
+	email: `string`,
+	selfieDocument: `string`,
+	identificationDocumentFront: `string`,
+	identificationDocumentBack: `string`,
+}
+```
+
+The `selfieDocument`, `identificationDocument` and `identificationDocumentBack` fields should be base64 encoded binary blobs representing images.
```
Comment (jophish) [inline fiatconnect-api.md:2266]: Can you provide some more detailed information on the fields required in this schema? (For example, how `phoneNumber` should be formatted, what the `idNumber` represents and how it should be formatted, etc.) It might seem a bit pedantic but it's important that we document the exact syntax of these fields.

**Comment by ailsoncgt** [comment]
@jophish I made a new commit

File: fiatconnect-api.md [hunk 3]
```diff
@@ -2233,6 +2234,40 @@ A KYC schema containing personal data about a user, as well as documents such as
...
+		isoRegionCode: `string`,
+		city: `string`,
+		postalCode?: `string`
+	},
+	phoneNumber: `string`,
+	idNumber: `string`,
+	email: `string`,
+	selfieDocument: `string`,
+	identificationDocumentFront: `string`,
+	identificationDocumentBack: `string`,
```
Comment (cajubelt) [inline fiatconnect-api.md:2262]: what kind of identity document is accepted? Any? Or just CPF cards? The description of the schema should probably go into this-- we're currently running into trouble with another KYC schema due to the ambiguity there on what document types are acceptable.

If it's CPF only, the schema should probably also be renamed.

Comment (ailsoncgt) [reply fiatconnect-api.md:2262]: @cajubelt the docs are those accepted as an official identification document in Brazil: `RG`, `CNH`, `RNM`

File: fiatconnect-api.md [hunk 4]
```diff
@@ -2233,6 +2234,41 @@ A KYC schema containing personal data about a user, as well as documents such as
 
 The `selfieDocument` and `identificationDocument` fields should be base64 encoded binary blobs representing images.
 
+#### 9.3.1.2. `PersonalDataAndDocumentsWithBack`
```
Comment (cajubelt) [inline fiatconnect-api.md:2237]: this schema name is misleading because it sounds more general than it actually is. Currently the constraints on document type, phone number and ID make this specific to Brazilian KYC checks.

File: fiatconnect-api.md [hunk 5]
```diff
@@ -2233,6 +2234,41 @@ A KYC schema containing personal data about a user, as well as documents such as
...
+
+A brasilian standard KYC schema containing personal data about a user, as well as documents such as an ID, photo and selfie.
+The accepted documents are [RG](https://en.wikipedia.org/wiki/Brazilian_identity_card), [CNH](https://en.wikipedia.org/wiki/Driving_licence_in_Brazil) and [RNM](https://en.wikipedia.org/wiki/Registro_Nacional_de_Estrangeiros)
```
Comment (cajubelt) [inline fiatconnect-api.md:2240]: all 3 of these documents have CPF on them, which the image would presumably include. Does idNumber still need to be its own field? Seems like it could be redundant, though I don't know the details of how you're doing KYC checks..

Comment (cajubelt) [reply fiatconnect-api.md:2240]: if you can drop the idNumber field, or clarify that it means these things specifically in the case where some field like `documentType` is equal to RG, CNH, or RNM, then you could potentially make this schema more general / reusable across regions

Comment (cajubelt) [reply fiatconnect-api.md:2240]: which could be helpful for other providers and consumers of the FC API

**Comment by ailsoncgt** [comment]
@cajubelt in my new commit remove `idNumber`. I confirmed that it is not actually required.
About the field you suggested, `documentType`, for our purpose it would be a dead field, because it will not be evaluated in real time for the KYC result. And without it, the scheme is open for other providers to receive other types of documents. I indicated the RG, RNM and CNH because they are the documents accepted in Brazil and consequently by our KYC provider. But the provider already recognizes the document type at the time of submission.

**Comment by jophish** [comment]
@ailsoncgt I believe the intent of the `documentType` field is to disambiguate, _for the client_ what kind of document is required. For example, if `documentType` could be one of `RG`, `RNM`, or `CNH`, then you, as a provider can set `allowedValues` to `["RG", "RNM", "CNH"]`, which indicates to the client that the user must submit one of those documents, and the client can then provide the specific means to the end-user to do so. If another provider wants to re-use this schema, but only supports, say, `RG`, then they can set their `allowedValues` field for `documentType` to just `["RG"]`. If _another_ provider comes along and wants to add support for a _new_ document type, they can do so by extending the list of valid document types and setting their `allowedValues` field as required.

TLDR, `documentType` is important in order to allow providers to extend this schema, so that clients can communicate to end-users exactly _what_ kind of document(s) the provider is requesting.

**Comment by cajubelt** [comment]
superceded by #92
